### PR TITLE
add recompile_thrift.sh to auto-generate replication.type.h

### PR DESCRIPTION
**Pull request by qinzuoyan**
- add recompile_thrift.sh to auto-generate replication.type.h, for #227
- add **_mutation_update**_ structure in replication.thrift, improve mutation marshall/unmarshall, for #182
- marshall/unmarshall error_code/task_code in string format,  for #180
- fix case-302.act, fix #234

### CODE REVIEW

**Comment by qinzuoyan** [comment]
@imzhenyu , please help review it.

File: include/dsn/cpp/auto_codes.h [hunk 1]
```diff
@@ -110,6 +110,11 @@ namespace dsn
             _internal_code = 0;
         }
 
+        task_code(dsn_task_code_t code)
+        {
+            _internal_code = code;
```
Comment (imzhenyu) [inline include/dsn/cpp/auto_codes.h:115]: this is dangerous as implicit cast from integer to task-code becomes possible. Better to use explicit method instead, e.g., set(xxx)

Comment (qinzuoyan) [inline include/dsn/cpp/auto_codes.h:115]: i see, and error_code is the same?

**Comment by imzhenyu** [comment]
Great! Will merge after test.

**Pull request merged**